src/core/Akka/Event/DeadLetterListener.cs

```diff
@@ -49,35 +51,235 @@ protected override void PostStop()
+                    }
+                    return true;
+                }
+                return false;
+            };
+        }
+
+        private Receive ReceiveWithSuspendLogging(TimeSpan suspendDuration)
+        {
+            return message =>
+            {
+                if (message is DeadLetter deadLetter)
+                {
+                    IncrementCount();
+                    if (_count == _maxCount)
+                    {
+                        var doneMsg = $", no more dead letters will be logged in next [{suspendDuration}]";
+                        LogDeadLetter(deadLetter.Message, deadLetter.Sender, deadLetter.Recipient, doneMsg);
+                        Context.Become(ReceiveWhenSuspended(suspendDuration, Deadline.Now + suspendDuration));
+                    }
+                    else
+                    {
+                        LogDeadLetter(deadLetter.Message, deadLetter.Sender, deadLetter.Recipient, "");
+                    }
+                    return true;
+                }
+                return false;
+            };
+        }
+
+        private Receive ReceiveWhenSuspended(TimeSpan suspendDuration, Deadline suspendDeadline)
+        {
+            return message =>
+            {
+                if (message is DeadLetter deadLetter)
+                {
+                    IncrementCount();
+                    if (suspendDeadline.IsOverdue)
+                    {
+                        var doneMsg = $", of which {_count - _maxCount - 1} were not logged. The counter will be reset now";
+                        LogDeadLetter(deadLetter.Message, deadLetter.Sender, deadLetter.Recipient, doneMsg);
+                        _count = 0;
+                        Context.Become(ReceiveWithSuspendLogging(suspendDuration));
+                    }
+                    return true;
+                }
+                return false;
+            };
+        }
+
+        private void LogDeadLetter(object message, IActorRef snd, IActorRef recipient, string doneMsg)
+        {
+            var origin = ReferenceEquals(snd, Context.System.DeadLetters) ? "without sender" : $"from {snd.Path}";
+            _eventStream.Publish(new Info(
+                recipient.Path.ToString(),
+                recipient.GetType(),
+                $"Message [{message.GetType().Name}] {origin} to {recipient.Path} was not delivered. [{_count}] dead letters encountered{doneMsg}. " +
+                $"If this is not an expected behavior then {recipient.Path} may have terminated unexpectedly. " +
+                "This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' " +
+                "and 'akka.log-dead-letters-during-shutdown'."));
+        }
+
+        /// <summary>
+        /// This class represents the latest date or time by which an operation should be completed.
+        /// </summary>
+        private class Deadline
```

Comment:
Temporary copied this from `Akka.Remote`. Maybe we should move it to `Akka.Utils` or something.

Reply:
Any reason why we shouldn't make this a `struct`? Any `deafult(Deadline)` calls anywhere in the main Akka.Remote code base? Or `null` checks?

Reply:
@Aaronontheweb I made the exercise of converting Deadline into a readonly struct, since this one is private to the DeadLetterListener. Take a look and let me know.

We could see whether this could also be done in `Akka.Remote`  in another PR.